test/distributed/test_c10d.py

```diff
@@ -1871,6 +1871,89 @@ def forward(self, x):
         return F.softmax(x, dim=1).to(dev0)
 
 
+@unittest.skipIf(TEST_WITH_TSAN, "TSAN is not fork-safe since we're forking in a multi-threaded environment")
+class DistributedDataParallelSingleProcessTest(TestCase):
```

Comment:
it has distributeddataparallel test in test_distributed.py,  do you think whether it is better  to put tests in test_distributed.py?

Reply:
Sure, let me move it. Would I be correct if I assume we plan to move all DDP related tests to `test_distributed.py`?

Reply:
yeah, I feel it is better to put ddp related tests in test_distributed.py, how do you think?

Reply:
Hey @zhaojuanmao, I compared `test_distributed.py` and `test_c10d.py`. If we plan to make them `test_ddp.py` and `test_c10d.py` in the future, I agree that this test better fits in `test_ddp.py`. But for now, it looks it fits more in `test_c10d.py`, as `DistributedDataParallelSingleProcessTest` is the single-process version of `DistributedDataParallelTest ` in `test_c10d.py` immediately after this test. And currently, a considerable part of `test_distributed.py` are testing APIs on c10d backend. Shall we keep this test in here for now, and we can certainly move it when we refactor the test file structures.